lib/metanorma/collection_manifest/collection_manifest.rb

```diff
@@ -0,0 +1,231 @@
+      config
+    end
+
+    def manifest_bibdata(config)
+      b = config.bibdata
+      @lang = b&.language&.first || "en"
+      @script = b&.script&.first || "Latn"
+    end
+
+    GUID = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+
+    def manifest_identifier(config, dir)
+      no_id = populate_id_from_doc(config, dir)
+      config.identifier =
+        if no_id && config.file # make file name be id
+          @collection.class.resolve_identifier(File.basename(config.file))
+        else
+          @collection.class.resolve_identifier(config.identifier)
+        end
+      Array(config.entry).each do |f|
+        manifest_identifier(f, dir)
+      end
+    end
+
+    def populate_id_from_doc(config, dir)
+      no_id = /^#{GUID}$/o.match?(config.identifier)
+      # GUID assumed to be no identifier supplied
+      if no_id && /\.xml$/.match?(config.file) &&
+          (i = retrieve_id_from_doc(config.file, dir))
+        config.identifier = i
+        no_id = false
+      end
+      no_id
+    end
+
+    def retrieve_id_from_doc(file, dir)
+      x = Nokogiri::XML(File.read(File.join(dir, file)), &:huge)
+      i = x.at("//xmlns:bibdata/xmlns:docidentifier[@primary = 'true']") ||
+        x.at("//xmlns:bibdata/xmlns:docidentifier")
+      i or return nil
+      @doctype ||= i["type"]&.downcase || "standoc"
+      load_isodoc
+      Util::key(@isodoc.docid_prefix(i["type"], i.text))
+    end
+
+    def load_isodoc
+      @isodoc and return
+      @collection.compile.load_flavor(@doctype)
+      @isodoc = Util::load_isodoc(@doctype)
+      @isodoc.i18n_init(@lang, @script, nil) # for @i18n.all_parts in docid
+    end
+
+    def manifest_sectionsplit(config, dir)
+      if config.sectionsplit && !config.file
+        config.sectionsplit = nil
+        Array(config.entry).each do |e|
+          e.attachment and next
+          e.sectionsplit = true
+        end
+      end
+      Array(config.entry).each do |f|
+        manifest_sectionsplit(f, dir)
+      end
+    end
+
+    def manifest_filexist(config, dir)
+      if config.file
+        file = @collection.class.resolve_fileref(dir, config.file)
+        @collection.class.check_file_existence(file)
+        config.file = Pathname.new(file).relative_path_from(Pathname.new(dir))
+      end
+      Array(config.entry).each do |f|
+        manifest_filexist(f, dir)
+      end
+    end
+
+    def manifest_expand_yaml(config, dir)
+      Array(config.entry).each do |e|
+        currdir = dir
+        /\.ya?ml$/.match?(e.file) and
+          currdir = manifest_expand_yaml_entry(e, dir)
+        manifest_expand_yaml(e, currdir)
+      end
+      config
+    end
+
+    def manifest_expand_yaml_entry(entry, dir)
+      f = @collection.class.resolve_fileref(dir, entry.file)
+      currdir = File.dirname(f)
+      @collection.class.check_file_existence(f)
+      entry.file = nil
+      entry.entry = CollectionConfig::Config.from_yaml(File.read(f)).manifest
+      if currdir != dir
```

Comment:
Path-awareness should be from within the CollectionManifest object. Make it track where it is. It was in the code I gave you...?

Reply:
Done